Code/Framework/AzCore/AzCore/DOM/DomPatch.cpp

```diff
@@ -0,0 +1,968 @@
+                        if (changedValueCount >= params.m_replaceThreshold)
+                        {
+                            AddPatch(PatchOperation::ReplaceOperation(path, after), PatchOperation::ReplaceOperation(path, before));
+                            return;
+                        }
+                    }
+                }
+            }
+
+            Path subPath = path;
+            for (size_t i = 0; i < afterSize; ++i)
+            {
+                if (i >= beforeSize)
+                {
+                    subPath.Push(PathEntry(PathEntry::EndOfArrayIndex));
+                    AddPatch(PatchOperation::AddOperation(subPath, after[i]), PatchOperation::RemoveOperation(subPath));
+                    subPath.Pop();
+                }
+                else
+                {
+                    subPath.Push(PathEntry(i));
+                    entriesToCompare.emplace(subPath, before[i], after[i]);
+                    subPath.Pop();
+                }
+            }
+
+            if (beforeSize > afterSize)
+            {
+                subPath.Push(PathEntry(PathEntry::EndOfArrayIndex));
+                for (size_t i = beforeSize; i > afterSize; --i)
+                {
+                    AddPatch(PatchOperation::RemoveOperation(subPath), PatchOperation::AddOperation(subPath, before[i - 1]));
+                }
+            }
+        };
+
+        auto compareNodes = [&](const Path& path, const Value& before, const Value& after)
+        {
+            if (before.GetNodeName() != after.GetNodeName())
+            {
+                AddPatch(PatchOperation::ReplaceOperation(path, after), PatchOperation::ReplaceOperation(path, before));
+            }
+            else
+            {
+                compareObjects(path, before, after);
+                compareArrays(path, before, after);
+            }
+        };
+
+        compareValues = [&](const Path& path, const Value& before, const Value& after)
+        {
+            if (before.GetType() != after.GetType())
+            {
+                AddPatch(PatchOperation::ReplaceOperation(path, after), PatchOperation::ReplaceOperation(path, before));
+            }
+            else if (before == after)
+            {
+                // If a shallow comparison succeeds we're pointing to an identical value or container
+                // and don't need to drill down.
+                return;
+            }
+            else if (before.IsObject())
+            {
+                compareObjects(path, before, after);
+            }
+            else if (before.IsArray())
+            {
+                compareArrays(path, before, after);
+            }
+            else if (before.IsNode())
+            {
+                compareNodes(path, before, after);
+            }
+            else
+            {
+                AddPatch(PatchOperation::ReplaceOperation(path, after), PatchOperation::ReplaceOperation(path, before));
+            }
```

Comment:
question: what case does this cover?

Reply:
If `before != after` (we do an equality check earlier) and they're not both the same container type, we do a replace, so it'll cover differing primitive values like numbers, strings, etc.